### ISSUE TITLE
Proguard/R8 issue on Windows with version 2.2.1

### ISSUE THREAD

**Issue by kaserdan** [Adyen/adyen-3ds2-android#27]
Hi,
I am having an issue with latest version 2.2.1 while building **obfuscated (Proguard/R8)** release version of our app on **Windows** CI build agent.
Build compiles just fine, but then some classes are missing and I keep getting exceptions such as:
```
Fatal Exception: java.lang.NoClassDefFoundError: Failed resolution of: Latd/S/a;
       at com.adyen.threeds2.internal.AppUpgradeBroadcastReceiver.onReceive(AppUpgradeBroadcastReceiver.java:1)
       at android.app.ActivityThread.handleReceiver(ActivityThread.java:3788)
       at android.app.ActivityThread.access$1400(ActivityThread.java:219)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1870)
       at android.os.Handler.dispatchMessage(Handler.java:107)
       at android.os.Looper.loop(Looper.java:214)
       at android.app.ActivityThread.main(ActivityThread.java:7356)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
```

I have noticed that latest version 2.2.1 is using uppercase letters in packagename (such as class metioned in exception - atd.S.a)
This is probably causing issues for Windows machines (case insensitive file paths). Building on Linux machine doesn't reproduce this issue.
Previous version (2.2.0) did not contain any uppercase letters in packagename and it works on Windows without issues.

**Comment by caiofaustino** [comment]
Hi @kaserdan, sorry for the delay, I'll do some tests on a Windows machine and see if I can reproduce the issue to investigate.

**Comment by caiofaustino** [comment]
Sorry I tested on Windows and everything seems to work fine, I don't think the issue is related to the upper case. Could you share some more information about your build? Are you using ProGuard/R8? Do you have Jettifier enabled?

**Comment by kaserdan** [comment]
Yep this issue is present only for our production release with Proguard/R8 turned on. Without proguard it is working fine. Jetifier is turned off.
We have tried to build production release on 3 different windows machines with same result - Fatal Exception: java.lang.NoClassDefFoundError
Proguard rules have properly listed all classes to keep them intact.
We have run into no issue on linux machine.
Is there any reason previous version did not have uppercase letters and this one does?

**Comment by kaserdan** [comment]
I will try to create sample project to replicate this issue

**Comment by caiofaustino** [comment]
Yeah we have updated from ProGuard to R8 in this version but I didn't see any issues during our tests.

Thanks, let me know if you find anything, we will be making changes to this in the near future as well so any input is appreciated.

**Comment by kaserdan** [comment]
Hm, I cannot replicate it with sample project neither. It seems it is caused in combination with something else in our project.
To better visualize the problem, please look at following:
https://github.com/kaserdan/ProguardIssueSample/blob/master/sample.png
The left part is classes.dex file classes from linux machine, the right part is Windows machine.
I appreciate your help. I will just keep digging to find the cause.

Is there any reason to compile with java 15? Android world is kind of behind with java and this is causing some issue e.g for jettifier.
What changes can we expect in future? Maybe we can just skip the version 2.2.1 and just use the latest version in near future.

**Comment by caiofaustino** [comment]
Yeah we made a mistake when updating a library and didn't notice it was compiled to java 15, we will fix that on the next release. I think you are safe skipping version 2.2.1 for now.

Please let me know if you still face any issues in future versions.

**Comment by caiofaustino** [comment]
Hi, this should be fixed in version 2.2.2, can anyone please check?
